src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/composite/AddContextDataToRead.java

```diff
@@ -0,0 +1,183 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms.composite;
+
+import com.google.cloud.dataflow.sdk.transforms.*;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGbkResult;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGroupByKey;
+import com.google.cloud.dataflow.sdk.transforms.join.KeyedPCollectionTuple;
+import com.google.cloud.dataflow.sdk.values.*;
+import com.google.common.collect.Lists;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReadContextData;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.RefAPIMetadata;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.RefAPISource;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.VariantsDataflowSource;
+import org.broadinstitute.hellbender.engine.dataflow.transforms.KeyReadsByUUID;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.util.List;
+import java.util.UUID;
+
+
+/**
+ * AddContextDataToRead pairs reference bases and overlapping variants with each GATKRead in the PCollection input.
+ * The variants are obtained from a local file (later a GCS Bucket). The reference bases come from the Google Genomics API.
+ *
+ * This transform is intended for direct use in pipelines.
+ */
+public class AddContextDataToRead {
+    public static PCollection<KV<GATKRead, ReadContextData>> add(PCollection<GATKRead> pReads, RefAPISource refAPISource, RefAPIMetadata refAPIMetadata, VariantsDataflowSource variantsDataflowSource) {
```

Comment:
Why pass a VariantsDataflowSource instead of the actual PCollection? The latter would be more flexible (plus it would prevent loading the variants twice if multiple functions do this).